blob/service.go

```diff
@@ -335,11 +398,18 @@ func (s *Service) getBlobs(
 	defer func() {
 		utils.SetStatusAndEnd(span, err)
 	}()
-	namespacedShares, err := s.shareGetter.GetSharesByNamespace(ctx, header, namespace)
-	if err != nil {
-		return nil, err
+
+	blobs := make([]*Blob, 0)
+	verifyFn := func(blob *Blob, _ Commitment) bool {
+		blobs = append(blobs, blob)
+		return false
 	}
-	return SharesToBlobs(namespacedShares.Flatten())
+
+	_, _, err = s.retrieve(ctx, header.Height(), namespace, nil, verifyFn)
```

Comment:
architectural nit: 
So verifyFn is generalized to allow hacking it and injecting custom behavior. The previous SharesToBlobs approach is much cleaner and does precisely what it should. Btw, SharesToBlob is not used, so we should either remove it or update it to support indexing so that we can keep using it her and avoid passing verifyFn. I recommend the latter.

Reply:
It was cleaner because we did not care about indexing. Indexing in making the logic more complicated even for getting all blobs

Reply:
SharesToBlobs function could be used from external packages in case someone needs it. I'd suggest to leave it.